### PR TITLE
Adds more gun/attachements options for players to use.

### PR DESCRIPTION
**Pull request by Wilchenx**
Cargo vendor:
~~Adds single shot grenade launchers~~
~~Adds tactical shotguns~~
Added tactical stocks for tactical shotguns
Increased amount of grenades avaible

Adds PPSH to cargo vendor under contraband
Adds ppsh ammo under contraband
Adds double barreled shotguns under contraband

Attachements:
Readded quickfire assemblies to marine SMG and Rifles
~~Flamers can now mount flamer attachements~~
Shotguns can now mount shotgun attachements
Added actual flashlight attachement to PPSH
Added ability to mount bipod to smartguns (allows for more defensive specialization on FOB duty)
~~Added ability to mount quickfire assembly to specialist grenade launcher~~

### CODE REVIEW

**Comment by Rohesie** [comment]
* Single-shot grenade launchers are MP weapons, usually with teargas.
They are pretty much useless to marines, as they can simply substitute them for UGLs, which can even carry more nades.
They don't even have a wielded sprite.

* Tactical shotguns are currently pretty OP. They can chain-stunlock xenos with slugs, and kick ass. In CM they are meta-protected behind armories, only to be used in code red. People who try to nab or distribute them get bwoinked for powergaming. I'd suggest a little rebalance before making those so available for the marines.

* Masterkey attachment on a shotgun sounds like a meme, but can actually bring important tactical benefits. Such as having both slugs and buckshot, being functional at both long and short range. I find the idea pretty interesting.

* Smoke nades serve for little to no purpose at the moment. Phosphorus could end up serving as a less-lethal alternative for riots, but realistically its use is close to null. Explosive grenades are the biggest noob bait, as they will easily cripple a squad if badly used. Distributing nades usually works more as a marine nerf than as aid, due to FF. All that said, what's the reasoning behind increasing their number?

* PPSh is a full meme gun. It's worse in every aspect than the regular SMG and it has no wielded-sprite. Why allowing attachments to it? I don't see an issue with this besides the lack of sprite, it won't impact game balance in any way, besides potentially being a small noob trap and being a little silly.

* Double-barrel shotguns are a bit annoying to handle but have the most kickass alpha strike, with no delay between its two shots. I'd keep them out until we fix the buckshot/PB issue first.

* QF was removed due to the QF+AP meta that gave marines victories 9 out of each 10 games. It made the game really boring. Though, admittedly, things have changed since then. I'd keep it out for now and first rework the attachments and guns.

* Flamer on flamer is full meme and serves no purpose, besides holding a little more burning fuel. Nothing against it.

* Interesting idea, the bipod on a smartgun. Did you test it? Works as intended?

* The fire_delay of the spec's grenade launcher was bugged for a bit in CM, and the specs could spam nades. It was hell and very broken, easily stunlocking any T3s and breaking absolutely any xeno groups. Please no.

**Comment by Rohesie** [comment]
*  Again, the Tactical Shotguns as they are offer no option over the pump shotguns. Instead they are simply superior due to firing twice as fast and requiring no reload. Please don't make them less scarce until we can actually give them a niche.

* Memery and LRP aside, the PPSh has no wielded sprite which both makes the game look uglier and offers less visual cues (it's important to know if someone is wielding a gun near you or not). Please get it a wielded sprite before you add it. It offers no relevant option and it's low-quality.

* Did you test the bipod on the smartgun? Does it work as intended?

**Comment by Wilchenx** [comment]
Bipods work on smartgun

**Comment by Wilchenx** [comment]
i should get a sprite for the ppsh from Keek tommorow

File: code/modules/projectiles/updated_projectiles/guns/flamer.dm [hunk 1]
```diff
@@ -20,14 +20,15 @@
...
+						/obj/item/attachable/magnetic_harness,
+						/obj/item/attachable/attached_gun/flamer) //welcome to marines, runner
 	flags_gun_features = GUN_UNUSUAL_DESIGN|GUN_WIELDED_FIRING_ONLY
 	gun_skill_category = GUN_SKILL_HEAVY_WEAPONS
 
 	New()
 		..()
 		fire_delay = config.max_fire_delay * 5
-		attachable_offset = list("rail_x" = 12, "rail_y" = 23)
+		attachable_offset = list("rail_x" = 12, "rail_y" = 23, "under_x" = 16, "under_y" = 16)
```
Comment (optimumtact) [inline code/modules/projectiles/updated_projectiles/guns/flamer.dm]: look at the state of his CODE :joy:

Comment (Wilchenx) [reply code/modules/projectiles/updated_projectiles/guns/flamer.dm]: what are you reffering to?

Comment (ninjanomnom) [reply code/modules/projectiles/updated_projectiles/guns/flamer.dm]: ///***relative pathing***///

**Comment by Rohesie** [comment]
Remove the attachments from the PPSh until it's field tested, the QF from the buffed SMG, and adjust the sprites and it's good to me.

File: code/game/machinery/vending/marine_vending.dm [hunk 2]
```diff
@@ -197,6 +197,7 @@
 					)
 
 	contraband = list(
+					/obj/item/ammo_magazine/smg/ppsh/extended = 20,
```
Comment (Rohesie) [inline code/game/machinery/vending/marine_vending.dm]: Check how the regular SMG is balanced regarding regular and extended ammo and put in similar or inferior values. Remember it scales with population anyway.

File: code/game/machinery/vending/marine_vending.dm [hunk 3]
```diff
@@ -552,6 +553,7 @@
 						/obj/item/attachable/stock/rifle = 3 ,
 						/obj/item/attachable/stock/revolver = 3,
 						/obj/item/attachable/stock/smg = 3,
+						/obj/item/attachable/stock/tactical = 4,
```
Comment (Rohesie) [inline code/game/machinery/vending/marine_vending.dm]: Same here. Make unusual stocks similar or inferior in numbers in comparison with the usual ones.

File: code/modules/projectiles/updated_projectiles/guns/smgs.dm [hunk 4]
```diff
@@ -216,6 +216,9 @@
 	origin_tech = "combat=3;materials=2;syndicate=4"
 	fire_sound = 'sound/weapons/smg_heavy.ogg'
 	current_mag = /obj/item/ammo_magazine/smg/ppsh
+	attachable_allowed = list(
+						/obj/item/attachable/flashlight)
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/guns/smgs.dm]: You did test how this looks, right?

Comment (Wilchenx) [reply code/modules/projectiles/updated_projectiles/guns/smgs.dm]: image

**Pull request merged**